src/prop/prop_proof_manager.h

```diff
@@ -43,12 +44,37 @@ class CDCLTSatSolver;
  */
 class PropPfManager : protected EnvObj
 {
+  friend class SatProofManager;
+
  public:
-  PropPfManager(Env& env,
-                context::UserContext* userContext,
-                CDCLTSatSolver* satSolver,
-                ProofCnfStream* cnfProof);
+  PropPfManager(Env& env, CDCLTSatSolver* satSolver, CnfStream& cnfProof);
+  /**
+   * Ensure that the given node will have a designated SAT literal that is
+   * definitionally equal to it.  The result of this function is that the Node
+   * can be queried via getSatValue(). Essentially, this is like a "convert-but-
+   * don't-assert" version of convertAndAssert().
+   */
+  void ensureLiteral(TNode n);
 
+  /**
+   * Converts a formula into CNF into CNF and asserts the generated clauses into
+   * the underlying SAT solver of d_cnfStream. Every transformation the formula
+   * goes through is saved as a concrete step in d_proof.
+   *
+   * The given formula has arbitrary Boolean structure via kinds AND, OR, EQUAL,
+   * XOR, IMPLIES. ITE and NOT. The conversion is done polynomially via Tseitin
+   * transformation, with the children of non-conjunctive kinds being abstracted
+   * as new literals, which are clausified with the respective "handle" methods
+   * below.
+
+   * @param node formula to convert and assert
+   * @param negated whether we are asserting the node negated
+   * @param removable whether the SAT solver can choose to remove the clauses
+   * @param input whether the node is from the input
+   * @param pg a proof generator for node
+   */
+  void convertAndAssert(
```

Comment:
I think it's cleaner to have a single class act to interact with. Otherwise it is more complicated in the prop engine to know which class to use.

Reply:
OK, let's go this way, even though I find it kinda ugly. But one thing to do now is to update the documentation so that there is not a complete duplication of the documentation in the prop proof manager and the proof cnf stream.

Reply:
I've updated the doc.